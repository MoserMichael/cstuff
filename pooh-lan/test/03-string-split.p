

text = 'the quick brown fox jumped over the lazy dog'

println( ~msg '*** split in for loop ***' )
for f split( ~string text )
  println( ~msg f )
end

text = 'the quick brown fox jumped over the lazy dog'
tokens = split( ~string text )

println( ~msg '*** split returning an array of tokens ***' )

for f tokens 
   println( ~msg f )
end

text = 'the:quick:brown:fox:jumped:over:the:lazy:dog twice'

println( ~msg '*** split with the : separator ***' )
for f split( ~string text ~separator ':' )
  println( ~msg f )
end
