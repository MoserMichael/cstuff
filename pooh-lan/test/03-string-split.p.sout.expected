*** split in for loop ***
the
quick
brown
fox
jumped
over
the
lazy
dog
*** split returning an array of tokens ***
the
quick
brown
fox
jumped
over
the
lazy
dog
*** split with the : separator ***
the
quick
brown
fox
jumped
over
the
lazy
dog twice
